Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/WMB.hs

```diff
@@ -6,51 +6,299 @@ module Domain.Action.UI.WMB
+            updatedAt = now
+          }
+  QVE.upsert vehicle
+  pure result
+
+getTripTransactionDetails ::
+  Domain.Types.TripTransaction.TripTransaction ->
+  Environment.Flow TripTransactionDetails
+getTripTransactionDetails trip = do
+  vehicleNumberHash <- getDbHash trip.vehicleNumber
+  mapping <-
+    VRME.findOneMapping vehicleNumberHash trip.routeCode
+      >>= fromMaybeM (InternalError "Vehicle Route mapping not found")
+
+  routeInfo <-
+    QRM.findByRouteCode trip.routeCode
+      >>= fromMaybeM (InternalError "Route not found")
+
+  let createStopInfo nameCodeObj point =
+        StopInfo
+          { name = nameCodeObj.stopName,
+            code = nameCodeObj.stopCode,
+            lat = Just point.lat,
+            long = Just point.lon
+          }
+
+  nameCodeObjStart <-
+    QRTS.findByLocation routeInfo.startPoint
+      >>= fromMaybeM (InternalError "No startPoint found")
+
+  nameCodeObjEnd <-
+    QRTS.findByLocation routeInfo.endPoint
+      >>= fromMaybeM (InternalError "No endPoint found")
+
+  let source = createStopInfo nameCodeObjStart routeInfo.startPoint
+  let destination = createStopInfo nameCodeObjEnd routeInfo.endPoint
+
+  pure
+    TripTransactionDetails
+      { tripTransactionId = trip.id.getId,
+        vehicleNum = trip.vehicleNumber,
+        vehicleType = mapping.vehicleServiceTierType,
+        source = source,
+        destination = destination
+      }
 
 getUiWmbTripActive ::
   ( ( Kernel.Prelude.Maybe (Kernel.Types.Id.Id Domain.Types.Person.Person),
       Kernel.Types.Id.Id Domain.Types.Merchant.Merchant,
       Kernel.Types.Id.Id Domain.Types.MerchantOperatingCity.MerchantOperatingCity
     ) ->
-    Environment.Flow API.Types.UI.WMB.TripLinkResp
+    Environment.Flow API.Types.UI.WMB.ActiveTripTransaction
   )
-getUiWmbTripActive = do error "Logic yet to be decided"
+getUiWmbTripActive (person, _, _) = do
+  checkFleetDriverAssociation person >>= \isAssociated -> unless isAssociated (throwError $ InternalError "Fleet-driver association not found")
+  driverId <- fromMaybeM (InternalError "Driver id not found") person
+  tripT <- QTTE.findAllTripTransactionByDriverIdStatus driverId (Just 1) (Just 0) (Just TRIP_ASSIGNED)
+  API.Types.UI.WMB.ActiveTripTransaction <$> case listToMaybe tripT of
+    Nothing -> pure Nothing
+    Just trip -> do
+      result <- getTripTransactionDetails trip
+      pure (Just result)
+
+-- haversineDistance :: LatLong -> LatLong -> Double
+-- haversineDistance (LatLong lat1 lon1) (LatLong lat2 lon2) =
+--   let toRadians deg = deg * pi / 180
+--       r = 6371.0
+--       dlat = toRadians (lat2 - lat1)
+--       dlon = toRadians (lon2 - lon1)
+--       a = sin (dlat / 2) * sin (dlat / 2) + cos (toRadians lat1) * cos (toRadians lat2) * sin (dlon / 2) * sin (dlon / 2)
+--       c = 2 * Kernel.Prelude.atan2 (sqrt a) (sqrt (1 - a))
+--    in r * c
+
+findClosestStop :: LatLong -> [(Text, LatLong)] -> Maybe Text
+findClosestStop loc stops =
```

Comment:
what if its equidistant from two stops ? we should consider stop with higher sequence number in that case.